agent/containermetadata/metadata_manager.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+const (
+	metadataEnvironmentVariable = "ECS_CONTAINER_METADATA_FILE"
+	inspectContainerTimeout     = 30 * time.Second
+)
+
+type MetadataStatus string
+
+const (
+	Metadata_NOT_READY = "NOT_READY"
```

Comment:
>> Why do we need INITIAL, READY, NOT_READY, etc? As a user, do I care about the state of the metadata?

> @petderek it'd have been better if you had just added to the threaded discussion than a comment here. Regardless, as a user, I'd care as it'd help me distinguish between "will i ever get this data" and "i'll eventually get this data if i keep polling"



Without split updates:
[ does data exist? ] => [ does the field I want exist? ] => [Yay, data!]

With:
[ does data exist? ] => [ is the 'ready' field set? ] => [ does the field i want exist? ] => [Yay, data!]

I'm not convinced that splitting up updates is simplifying. Either way, you will still need to poll for metadata. However, now anyone calling this needs to be aware of a different field that guarantees completeness.